Add pak loader

Tadinu opened the pull request:
* `ERRFileType` add PAK
* Add `URRPakLoader`, loading pak files (compressed cooked assets) from `/Game/RapyutaContents/DynamicContents/Paks`, only used in packaged build
* Move EMPTY_STR from ActorCommon to RRGeneralUtils.h to be more decoupling
* Move `ERRFileType & ERRShapeType` from ActorCommon -> ObjectCommon as usable by any generic object
* `URRAssetUtils::SavePackageToAsset()` clear `RF_Transient`, which is not allowed for an object being saved to uasset
* `URRCoreUtils::LoadImageToTexture()` add `bInSaveToAsset`, using `ImportObject()` to import texture if to be saved to uasset, since `FImageUtils::ImportFileAsTexture2D()` only creates a transient non-source-data texture2d that is ineligible to be saved to uasset.

Commit:
ERRFileType add PAK
Move ERRFileType & ERRShapeType from ActorCommon -> ObjectCommon
URRAssetUtils::SavePackageToAsset() clear RF_Transient

Commit:
RRCoreUtils::LoadImageToTexture() add bInSaveToAsset,
using ImportObject built-in api to import texture if to be saved to uasset

Commit:
Add RRPakLoader
Move EMPTY_STR to RRGeneralUtils.h

Commit:
RRCoreUtils add RunMonitoredProcess()
FRRRobotModelData add HasDriveComponents()

yuokamoto commented on Source/RapyutaSimulationPlugins/Private/Core/RRGameSingleton.cpp:
```diff
@@ -97,6 +100,33 @@ bool URRGameSingleton::InitializeResources(bool bInRequestResourceLoading)
+        if (PakLoader->Initialize())
+        {
+            // Load pak files
+            for (const auto& paksBasePath : GetDynamicAssetsBasePathList(ERRResourceDataType::UE_PAK))
```
Do we need to load packaged asset?
It causes load all asset in the begining and make start up time slower and slower

I think this is fine as a first step but should stat thinking about loading as demands. we maybe define list of initial loading asset DefaultConfig.ini

Tadinu replied:
So this is already added as backlog task to be done.
Basically Sim should load only resources of entity models specified by users, but we need to define configs for them, similar to Import~ModelNameList arg of ImportEntitiesCmdlet.
When users define it, probably in INI, only resource assets (uncooked and Paks) that correspond to stipulated models are loaded.

yuokamoto reviewed (commented):
Do we need both `UE_PAK_MOUNTED_BASE_FOLDER_NAME` and `FOLDER_PATH_ASSET_PAKS` ?

Commit:
RobotModelData add IsRobotModel()/IsObjectModel()

Commit:
IsRobotModel() use JointPropListNum also

yuokamoto commented on Source/RapyutaSimulationPlugins/Public/Core/RRCoreUtils.h:
```diff
@@ -613,6 +617,31 @@ class RAPYUTASIMULATIONPLUGINS_API URRCoreUtils : public UBlueprintFunctionLibra
+            // Already slept in [Update()]
+            if (URRCoreUtils::GetElapsedTimeSecs(lastMarkedTime) > InTimeOutSecs)
+            {
+                InProcess->Cancel();
```
Don't you need break while loop or Update() will return false?

Tadinu replied:
Yeah I thought about it but then figured out the api actually already handles it more properly to make sure the while loop here only breaks after the process is terminated, specifically
* [Cancel() ](https://github.com/EpicGames/UnrealEngine/blob/release/Engine/Source/Runtime/Core/Public/Misc/MonitoredProcess.h#L71)-> ON `Canceling`
* [FMonitoredProcess::TickInternal()](https://github.com/EpicGames/UnrealEngine/blob/release/Engine/Source/Runtime/Core/Private/Misc/MonitoredProcess.cpp#L135) -> upon cancelling turns OFF `bIsRunning` (which is of atomic type thus thread-safe) **only after officially terminating the process**, also making our while loop here break accordingly

Tadinu commented:
yeah UE_PAK_MOUNTED_BASE_FOLDER_NAME was only for debug, removed

Commit:
add log upon process timeout
WaitUntilThenAct use GetElapsedTimeInSec()

yuokamoto commented on Source/RapyutaSimulationPlugins/Private/Core/RRCoreUtils.cpp:
```diff
@@ -261,18 +261,16 @@ bool URRCoreUtils::WaitUntilThenAct(TFunctionRef<bool()> InCond,
-        elapsed_time = FTimespan(FDateTime::UtcNow() - begin).GetTotalSeconds();
     }
-    // Either InCond() is met or [elapsed_ticks] is over [InTimeoutInSec]
+    // Either InCond() is met or [ElapsedTime] is over [InTimeoutInSec]
```
[ElapsedTime] looks like variable

Commit:
RRGameMode Print USING_THREAD_SANITISER
Reenable PrintSimConfig() which shows essential Engine & user INI configs
Rem obsolete physxDispatcher cmdline param
